cachecontrol/controller.py

```diff
@@ -117,6 +117,12 @@ def cached_request(self, request):
             return resp
 
         headers = CaseInsensitiveDict(resp.headers)
+        if not headers or 'date' not in headers:
+            # With date or etag, the cached response can never be used
+            # and should be deleted.
+            if 'etag' not in headers:
```

Comment:
I'm not sure this logic is correct. An Etag should be enough to cache, even if there is not a date because the client can still send a conditional request. I think the test should be `'etag' not in headers and 'date' not in headers`. Does that makes sense? Let me know if I'm misunderstanding.